src/main/java/com/netflix/blitz4j/LoggingContext.java

```diff
@@ -172,6 +182,37 @@ public LocationInfo generateLocationInfo(LoggingEvent event) {
         return locationInfo;
     }
 
+    private boolean isUsingNFPatternLayout(LoggingEvent event) {
+        final Category logger = event.getLogger();
+        return logger != null && isUsingNFPatternLayout(logger.getAllAppenders());
+
+    }
+
+    private boolean isUsingNFPatternLayout(Enumeration enumeration) {
```

Comment:
I feel like this could be cached somehow instead of having to enumerate all patterns for every log event.